cargo-concordium/src/main.rs

```diff
@@ -893,35 +931,91 @@ fn handle_run_v1(run_cmd: RunCommand, module: &[u8]) -> anyhow::Result<()> {
                         };
 
                         match func_schema_opt {
+                            // V2 schemas don't have schemas for events.
                             Some(func_schema) => (
                                 true,
                                 func_schema.parameter(),
                                 func_schema.return_value(),
                                 func_schema.error(),
+                                None,
                             ),
-                            None => (true, None, None, None),
+                            None => (true, None, None, None, None),
                         }
                     }
-                    None => (true, None, None, None),
+                    None => (true, None, None, None, None),
+                }
+            }
+            Some(VersionedModuleSchema::V3(module_schema)) => {
+                // Getting the event schema.
+                let schema_event = match module_schema.contracts.get(contract_name) {
+                    Some(contract_schema) => contract_schema.event(),
+                    None => None,
+                };
+
+                match module_schema.contracts.get(contract_name) {
```

Comment:
You are getting the contract schema twice in a row here, maybe call the `event()` method in this match instead of the above